Tree/js/index.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Clear tree and build again
+ */
+function clearAndBuild() {
+  $('.tree').empty();
+
+  elements.forEach((element) => {
+    if (element.level > max) {
+      max = element.level;
+    }
+  });
+
+  building();
+}
+
+/**
+ * Cleans the tree and rebuild
+ * @param {number} currentLevel
+ * @param {string} parent
+ */
+function building(currentLevel, parent) {
+  currentLevel = (currentLevel) ? currentLevel : 0;
+  parent = (parent) ? parent : 'tree';
+
+  elements.forEach((element) => {
+    if (element.level === currentLevel && parent === element.parentId) {
+      const ul = $('<ul/>');
+      const li = $('<li/>');
+      const content = $('<div/>');
+      const arrow = $('<div/>');
+      const image = $('<div/>');
+      const name = $('<div/>').text(element.name);
+
+      ul.addClass('opened');
+      ul.addClass(`${element.id}`);
+      content.addClass(`${element.type}`);
+      name.addClass('name');
+      image.addClass(`${element.type}-image image`);
+      arrow.addClass(`${element.type}-open image toggle`);
+      arrow.addClass(`${element.id}-toggle`);
+
+      content.append(arrow, image, name);
+      li.append(content, ul);
+
+      if ($(`.${parent}`).is('ul')) {
+        $(`.${element.parentId}`).append(li);
+      } else {
+        ul.append(li);
+        $(`.${element.parentId}`).append(ul);
+      }
+
+      $(`.${element.id}-toggle`).on('click', () => {
```

Comment:
Also you can create selector, like .selector, set data params for this selector:

```$('.selector').data({})```
and add event to this selector:
```
$(document).on('click', `.selector`, callback);
```
In the callback you can get data for this selector.